socketio/src/client/raw_client.rs

```diff
@@ -288,8 +288,29 @@ impl RawClient {
                 }
 
                 if let Some(ref attachments) = socket_packet.attachments {
-                    if let Some(payload) = attachments.get(0) {
-                        ack.callback.deref_mut()(Payload::Binary(payload.to_owned()), self.clone());
+                    if let Some(payload) = attachments.first() {
+                        ack.callback.deref_mut()(
+                            Payload::Binary(payload.to_owned()),
+                            self.clone(),
+                        );
+                    }
+                    if ack.time_started.elapsed() < ack.timeout {
+                        if let Some(ref payload) = socket_packet.data {
+                            ack.callback.deref_mut()(
+                                Payload::from(payload.to_owned()),
+                                self.clone(),
+                            );
+                        }
+                        if let Some(ref attachments) = socket_packet.attachments {
+                            if let Some(payload) = attachments.first() {
+                                ack.callback.deref_mut()(
+                                    Payload::Binary(payload.to_owned()),
+                                    self.clone(),
+                                );
+                            }
+                        }
+                    } else {
+                        // Do something with timed out acks?
```

Comment:
Definitely! Remove  the ack for sure, also call the error callback? Or maybe just ignore. Would be good to see what the JS implementation does.